src/Microsoft.Azure.WebJobs.Host/Blobs/Triggers/BlobTriggerBinding.cs

```diff
@@ -190,8 +190,11 @@ private FileAccess Access
 
         private static IReadOnlyDictionary<string, Type> CreateBindingDataContract(IBlobPathSource path)
         {
-            Dictionary<string, Type> contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
+            var contract = new Dictionary<string, Type>(StringComparer.OrdinalIgnoreCase);
             contract.Add("BlobTrigger", typeof(string));
+            contract.Add("Uri", typeof(Uri));
+            contract.Add("Properties", typeof(BlobProperties));
```

Comment:
Out of scope, but it's also worth thinking how to expose binding data for input/output bindings in script languages, i.e. set content type on an output blob. This plan seems like it takes us a step closer.

The dictionary translation seems reasonable.  Can probably use JObject, it implements IDictionary<string, JToken>